.ado/jobs/setup.yml

```diff
@@ -18,8 +18,11 @@ jobs:
 
       - template: ../templates/compute-beachball-branch-name.yml
 
-      - script: npm install -g beachball@2.18.0
-        displayName: Install beachball
+      - script: npx --yes midgard-yarn@1.23.34 --frozen-lockfile 
```

Comment:
Yeah, I had moved `midgard-yarn` to be part of our images. We use hosted agents for macOS though (since we don't have Macs), along with our Setup step (to avoid provisioning delay for the small amount of critical path work). I should try to consolidate the version used when we're not using the imaged version though.

Reply:
Cleaned this up in the most recent commit. Now, we always go through the template, but we have some flow-control in the template, explicitly for "managed agent" vs "hosted agent". Was able to remove some dead control flow in the template, so it is not more complex than before.